parser/prism/Translator.cc

```diff
@@ -76,6 +76,71 @@ template <typename... TArgs>
     Exception::raise(reasonFormatStr, forward<TArgs>(args)...);
 }
 
+// Helper function to check if an AST expression is a string literal
+bool isStringLit(ast::ExpressionPtr &expr) {
+    if (auto lit = ast::cast_tree<ast::Literal>(expr)) {
+        return lit->isString();
+    }
+    return false;
+}
+
+// Helper function to merge multiple string literals into one
+ast::ExpressionPtr mergeStrings(core::MutableContext &ctx, core::LocOffsets loc,
```

Comment:
```suggestion
ast::ExpressionPtr mergeStrings(core::MutableContext ctx, core::LocOffsets loc,
```

Contexts are always passed by value. They contain a `GlobalState &` inside them so that they do not need to copy the full GlobalState. But accepting a `MutableContext &` means that this method can do

```
ctx = compute_new_context()
```

and overwrite the caller's context. That's ~never what you want.